### DISCUSSION TITLE
How do I see if the max sequence length was reached instead of the stop token?

### DISCUSSION THREAD

**Discussion by PhilipMay** [vllm-project/vllm#5192] [Q&A]
Hi,
when I use the OpenAI API I get a return value called `finish_reason`.
It can have the value of `stop` if the last token was the stop token or
the value of `length` means the API stopped the completion because of running into a token limit.

How do I see if the stop token was returned or not when I use vLLM?
How do I see if the token limit was reached?

**Comment by PhilipMay** [comment]
Can anybody help me please? Many thanks?

**Reply by g-eoj** [reply]
Have you tried the OpenAI Compatible Server? https://docs.vllm.ai/en/stable/serving/openai_compatible_server.html

**Reply by PhilipMay** [reply]
I do not want to use vLLM with the RESTful API but with the Python API.
The OpenAI Compatible Server seems to be focused on RESTful API only.
